karapace/schema_reader.py

```diff
@@ -263,7 +257,7 @@ def _is_ready(self) -> bool:
             return False
         # Offset in the response is the offset for the next upcoming message.
         # Reduce by one for actual highest offset.
-        self._highest_offset = list(offsets.values())[0] - 1
+        self._highest_offset = end_offset - 1
```

Comment:
same here, probably its worth checking if its really right and how its handled for an empty topic

Reply:
Related explanation in the backups code: https://github.com/Aiven-Open/karapace/pull/781